packages/git/src/index.ts

```
@@ -59,6 +64,12 @@ async function getChangedFilesSince({
   const divergedAt = await getDivergedCommit(cwd, ref);
   // Now we can find which files we added
   const cmd = await spawn("git", ["diff", "--name-only", divergedAt], { cwd });
+  if (cmd.code !== 0) {
```

when this could happen if you throw an error already in `getDivergedCommit`?

---

I agree if we've already confirmed that the given ref exists, then the likelihood that this command fails is low.

But there are some `git` configurations which can still make `git diff` return non-zero errors when it's run, if for example, they have whitespace checking enabled (https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---check) then `git diff` could return with a non-zero exit code.

I guess it's still good practice to handle the case when shelling out to `git` may unexpectedly fail, rather than silently allowing the error. :)

---

U are probably right. Its a scary async world out there and better be safe than sorry 😅